lib/apiservers/portlayer/restapi/utils/utils.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/tasks"
+	"github.com/vmware/vic/pkg/uid"
+)
+
+func GetOperationID(params interface{}) string {
+	var opID *string
+	switch v := params.(type) {
```

Comment:
Do we need to filter by the specific operation being called? My preference would be for a general mechanism and we limit it's use implicitly based on which Handlers we call the function from.

If we do need this, the switch will not function as I think intended - you need an explicit `fallthrough` statement in Go.